File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
        int count = 0;
        for(int i=0;i<s.length();i++){
            char ch = s.charAt(i);
                
            if(!st.isEmpty()){
                if(ch==')' && st.peek()=='(' || (ch==']' && st.peek()=='[') || (ch=='}' && st.peek()=='{')){
                    st.pop();
                }
                else{
                    st.push(ch);
                }
            }else{
                st.push(ch);
            }
        
        }
        // if(!st.isEmpty()){
        //     return false;
        // }
        return st.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = []
        for ch in s:
            if st:
                if ch == ')' and st[-1]=='(':
                    st.pop()
                elif ch == ']' and st[-1]=='[':
                    st.pop()
                elif ch == '}' and st[-1]=='{':
                    st.pop()
                else:
                    st.append(ch)
            else:
                st.append(ch)
        if st:
            return False
        return True